lib/notifications/local/local_notification_manager.dart

```diff
@@ -0,0 +1,71 @@
+import 'dart:io';
+
+import 'package:firebase_messaging/firebase_messaging.dart';
+import 'package:flutter_local_notifications/flutter_local_notifications.dart';
+import 'package:flutter_template/log/log.dart';
+import 'package:flutter_template/notifications/data/data_notification_manager.dart';
+
+/// Manages local notifications and their details.
+///
+/// When used handle the click on the notifications here.
+/// To obtain an instance use `serviceLocator.get<LocalNotificationsManager>()`
+class LocalNotificationsManager {
```

Comment:
This component should be independent from any FCM code, including RemoteMessage.

Also, it might not need the DataNotificationsManager but it's OK if it uses it since it's a higher lvl component. Without it, it can handle the notifications itself.